src/mono/wasm/runtime/startup.ts

```diff
@@ -628,6 +631,56 @@ function downloadResource(request: ResourceRequest): LoadingResource {
     };
 }
 
+export function resolve_module_url(behavior: AssetBehaviours) {
+    let asset: AssetEntry | undefined = config.assets?.find(a => a.behavior == behavior);
+    if (!asset) {
+        switch (behavior) {
+            case "js-module-crypto":
+                asset = {
+                    name: "dotnet-crypto-worker.js",
+                    behavior
+                };
+                break;
+            case "js-module-threads":
+                asset = {
+                    name: "dotnet.worker.js",
+                    behavior
+                };
+                break;
+            default:
+                throw new Error("Not a module" + behavior);
+        }
+    }
+    if (!asset.resolvedUrl) {
+        asset.resolvedUrl = resolve_url(asset, "");
+    }
+    return asset;
+}
+
+function resolve_url(asset: AssetEntry, sourcePrefix: string): string {
```

Comment:
Is this function expected to always return a valid url?

Reply:
on node on windows it returns `C:/some/path/file.xxx` on browser `http://some.net/path/file.xxx` on V8 `./file.xxx`